dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/service/RecordTransformer.java

```diff
@@ -0,0 +1,309 @@
+        }
+
+    }
+
+    /**
+     * Generates an entity class from a record class.
+     *
+     * The generated entity class will have a public field, getter method, and setter method
+     * for each record component.
+     *
+     * The generated entity class will have a public no-args constructor, a public constructor
+     * that accepts the corresponding record, and a public toRecord() method that returns a
+     * new instance of the record.
+     *
+     * Initially copied from @nmittles pull #25248
+     *
+     * @param recordClass     - the record class that will be transformed into an entity class
+     * @param entityClassName - the name of the entity class
+     *
+     * @return the byte array representation of the generated entity class
+     */
+    public static byte[] generateEntityClassBytes(Class<?> recordClass, String entityClassName) {
+        final boolean trace = TraceComponent.isAnyTracingEnabled();
+
+        final String CTOR = "<init>";
+        final String TO_RECORD = "toRecord";
+
+        // Names of classes used in generating an entity class from a record
+        final String _recordClassName = recordClass.getName().replace(".", "/");
+        final String _recordClassNameDesc = org.objectweb.asm.Type.getDescriptor(recordClass);
+
+        final String _entityClassName = entityClassName.replace('.', '/');
+        final String _objectClassName = Object.class.getName().replace('.', '/');
+
+        final String _sourceFileName = entityClassName.substring(entityClassName.lastIndexOf(".") + 1) + ".java";
+
+        if (recordClass.getTypeParameters().length != 0) {
+            throw new IllegalStateException("The " + recordClass.getName() + " record has one or more generic types " +
+                                            Arrays.asList(recordClass.getTypeParameters()) + ". This record class cannot be transformed into "
+                                            + " and entity class since entity classes cannot be generic."); //TODO NLS
+        }
+
+        // Collect record component information, must preserve order
+        Map<String, RecordComponentInfo> recordComponents = new LinkedHashMap<>();
+        for (RecordComponent component : recordClass.getRecordComponents()) {
+            RecordComponentInfo info = new RecordComponentInfo(component);
+            RecordComponentInfo previous = recordComponents.put(info.getMethodName(""), info);
+
+            if (previous != null) {
+                throw new IllegalStateException("The " + recordClass.getName() + " record has two components [" +
+                                                info.name + ", " + previous.name + "] that cannot be tranformed into an entity class "
+                                                + "because the entity class getter and setter methods would be ambiguious."); //TODO NLS
```

Comment:
Similar to above. Maybe use DataException or MappingException for this one?
```suggestion
                throw new DataException("The " + recordClass.getName() + " record has two components [" +
                                                info.name + ", " + previous.name + "] that cannot be tranformed into an entity class "
                                                + "because the entity class getter and setter methods would be ambiguous."); //TODO NLS
```